js/main.js

```diff
@@ -116,25 +154,125 @@ const createPin = (array) => {
   return pinElement;
 };
 
-const createNodeFragment = (pin) => {
+const translateHouseType = (objectValue) => {
+  switch (objectValue) {
+    case `flat`:
+      objectValue = `Квартира`;
+      break;
+    case `bungalow`:
+      objectValue = `Бунгало`;
+      break;
+    case `house`:
+      objectValue = `Дом`;
+      break;
+    case `palace`:
+      objectValue = `Дворец`;
+      break;
+  }
+
+  return objectValue;
+};
+
+const createPhotosElements = (photosArr, source) => {
+  for (let i = 0; i < photosArr.length; i++) {
+    const photoElement = cardPhotoTemplate.cloneNode(true);
+    photoElement.src = photosArr[i];
+    source.appendChild(photoElement);
+  }
+  return source;
+};
+
+const createFeaturesElements = (featuresArr) => {
+  const mapCardNode = mapCardTemplate.cloneNode(true);
+  const featuresNode = mapCardNode.querySelector(`.popup__features`);
+  const featuresElement = featuresNode.querySelectorAll(`.popup__feature`);
+
+  if (featuresArr.length === 0) {
+    for (let featureElement of featuresElement) {
+      delChild(featuresNode, featureElement);
+    }
+  } else {
+    for (let i = 0; i < featuresArr.length; i++) {
+      if (!featuresArr.includes(`wifi`)) {
+        featuresNode.querySelector(`.popup__feature--wifi`).style = `display: none`;
+      }
+      if (!featuresArr.includes(`dishwasher`)) {
+        featuresNode.querySelector(`.popup__feature--dishwasher`).style = `display: none`;
+      }
+      if (!featuresArr.includes(`parking`)) {
+        featuresNode.querySelector(`.popup__feature--parking`).style = `display: none`;
+      }
+      if (!featuresArr.includes(`washer`)) {
+        featuresNode.querySelector(`.popup__feature--washer`).style = `display: none`;
+      }
+      if (!featuresArr.includes(`elevator`)) {
+        featuresNode.querySelector(`.popup__feature--elevator`).style = `display: none`;
+      }
+      if (!featuresArr.includes(`conditioner`)) {
+        featuresNode.querySelector(`.popup__feature--conditioner`).style = `display: none`;
+      }
+    }
+  }
+
+  return featuresNode;
+};
+
+const createCard = (object) => {
```

Comment:
```suggestion
const createCard = (dataObject) => {
```
Лучше явно указывать что за объект, иначе быстро забывается и приходится возвращаться чтобы освежить память